x-pack/plugins/observability_solution/observability/public/components/alert_search_bar/containers/use_alert_search_bar_state_container.tsx

```diff
@@ -162,43 +161,35 @@ function setupUrlStateSync(
   });
 }
 
-function syncUrlStateWithInitialContainerState(
+function initializeUrlAndStateContainer(
   timefilterService: TimefilterContract,
   stateContainer: AlertSearchBarStateContainer,
   urlStateStorage: IKbnUrlStateStorage,
-  urlStorageKey: string,
-  replace: boolean = true
+  urlStorageKey: string
 ) {
   const urlState = alertSearchBarState.decode(
     urlStateStorage.get<Partial<AlertSearchBarContainerState>>(urlStorageKey)
   );
+  const validUrlState = isRight(urlState) ? pipe(urlState).right : {};
+
+  const { from, to } = timefilterService.getTime();
+  const timeFilterState = timefilterService.isTimeTouched()
+    ? {
+        ...defaultState,
+        rangeFrom: from,
+        rangeTo: to,
+      }
+    : {};
 
-  if (isRight(urlState)) {
-    const newState = {
-      ...defaultState,
-      ...pipe(urlState).right,
-    };
+  const currentState = {
+    ...defaultState,
+    ...timeFilterState,
+    ...validUrlState,
+  };
 
-    stateContainer.set(newState);
-    urlStateStorage.set(urlStorageKey, stateContainer.get(), {
-      replace: true,
-    });
-    return;
-  } else if (timefilterService.isTimeTouched()) {
-    const { from, to } = timefilterService.getTime();
-    const newState = {
-      ...defaultState,
-      rangeFrom: from,
-      rangeTo: to,
-    };
-    stateContainer.set(newState);
-  } else {
-    // Reset the state container when no URL state or timefilter range is set to avoid accidentally
-    // re-using state set on a previous visit to the page in the same session
-    stateContainer.set(defaultState);
-  }
-
-  urlStateStorage.set(urlStorageKey, stateContainer.get(), {
-    replace,
+  stateContainer.set(currentState);
+  urlStateStorage.set(urlStorageKey, currentState, {
+    replace: true,
   });
+  urlStateStorage.kbnUrlControls.flush();
```

Comment:
Here is the main change: after calculating the initial state, we update both the URL and state container accordingly and then start the syncing process on line 133.